src/Repository/Entity/Repository.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+namespace OAT\DependencyResolver\Repository\Entity;
+
+class Repository implements \JsonSerializable
+{
+    public const CSV_TITLES = [
+        'repositoryName',
+        'extensionName',
+        'composerName',
+        'privacy',
+        'packagist',
+        'defaultBranch',
+    ];
+
+    /** @var bool */
+    private $analyzed;
+
+    /** @var string */
+    private $owner;
+
+    /** @var string */
+    private $name;
+
+    /** @var bool */
+    private $private;
+
+    /** @var string */
+    private $defaultBranch;
+
+    /** @var string */
+    private $extensionName;
+
+    /** @var string */
+    private $composerName;
+
+    /** @var bool */
+    private $onPackagist;
+
+    /** @var RepositoryBranch[]|array */
+    private $branches;
+
+
+
+    public function __construct(
```

Comment:
I'm not sure about the default values here.... I don't think that a repository can exist without an owner, or name, or at least one branch.... etc. You should not allow default value where it doesn't make sense. What do you think?